### PR TITLE
Update the pull request template with a security-related section

### PR DESCRIPTION
**Pull request by ivoanjo**
**What does this PR do?**:

This PR updates the pull request template with a security-related question, so we make sure folks remember to loop in the correct team if the PR can have an impact on "code that signs or publishes builds or packages, or handles credentials of any kind".

**Motivation**:

We're moving to standardize this on all Datadog libraries.

For Datadog employees, you can find the discussion on the internal "RFC: Security review for ci/cd changes in public repos" RFC.

**Additional Notes**:

N/A

**How to test the change?**:

Once this is merged, open a new PR, validate that the new items show up.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -5,19 +5,26 @@ for guidance on how to set up your development environment,
 run the test suite, write new integrations, and more.
 -->
 
-**What does this PR do?**
+**What does this PR do?**:
```
Comment (ekump) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: Grammar nitpick: is having a colon after a question mark correct? Should the title be re-phrased `What this PR does: `

**Review by ekump — APPROVED** [review]
Couple of minor grammar questions, but nothing blocking

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (ivoanjo) [reply .github/PULL_REQUEST_TEMPLATE.md]: :+1: Fixed in 6ae39459d338bef1b9d936aaf471bca75db4f235

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -5,19 +5,26 @@ for guidance on how to set up your development environment,
...
-**Motivation**
+**Motivation**:
 <!-- What inspired you to submit this pull request? -->
 
-**Additional Notes**
+**Additional Notes**:
 <!-- Anything else we should know when reviewing? -->
 
-**How to test the change?**
+**How to test the change?**:
```
[the remark this answers is not in the document]
Comment (ivoanjo) [reply .github/PULL_REQUEST_TEMPLATE.md]: :+1: Fixed in 6ae39459d338bef1b9d936aaf471bca75db4f235

**Comment by ivoanjo** [comment]
I was trying out if the merge queue thing would work for our repo, but I guess it depends on some missing setup.

So, old-skool manual merge it is! Thanks Edmund for the review.

**Pull request merged**